backend/inspirehep/search/api.py

```diff
@@ -127,10 +127,13 @@ class Meta:
     def execute(self, *args, **kwargs):
         results = super().execute(*args, **kwargs)
         if not results.hits and request:
-            query_string = request.values.get("q", "", type=str)
-            reference_match = self.match_reference(query_string)
-            if reference_match:
-                return reference_match
+            try:
+                query_string = request.values.get("q", "", type=str)
+                reference_match = self.match_reference(query_string)
+                if reference_match:
+                    return reference_match
+            except Exception:
+                LOGGER.exception("Match reference error.", query_string=query_string)
```

Comment:
Isn't better to raise here some RestError? so frontend will receive some error which can be displayed to the user later? Otherwise user will get empty results.

Reply:
This is exactly what we want to avoid :)